omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+            )  # shape (N_samples, n_spheres)
+
+            # Non-zero distances correspond to a collision detection (or close to a collision, within activation_distance
+            # So valid collision-free samples are those where max(n_obs_spheres) == 0 for a given sample
+            collision_results = dist.max(dim=-1).values != 0
+
+        # Return results
+        return collision_results  # shape (B,)
+
+    def compute_trajectory(
+        self,
+        target_pos,
+        target_quat,
+        is_local=False,
+        max_attempts=5,
+        enable_finetune_trajopt=True,
+        return_full_result=False,
+        success_ratio=None,
+        attached_obj=None,
+    ):
+        """
+        Computes the robot joint trajectory to reach the desired @target_pos and @target_quat
+
+        Args:
+            target_pos ((N,3)-tensor): (N, 3) or (3,)-shaped tensor, where N is expected to be self.batch_size, and each
+                entry is (x,y,z) position to reach. If only a single (3,) array is given but N > 1, it will
+                be broadcast to create the appropriate array.
+            target_quat ((N,4)-tensor): (N, 4) or (4,)-shaped tensor, where N is expected to be self.batch_size, and each
+                entry is (x,y,z,w) quaternion orientation to reach. If only a single (3,) array is given but N > 1,
+                it will be broadcast to create the appropriate array.
+            is_local (bool): Whether @target_pos and @target_quat are specified in the robot's local frame or the world
+                global frame
+            max_attempts (int): Maximum number of attempts for trying to compute a valid trajectory
+            enable_finetune_trajopt (bool): Whether to enable timing reparameterization for a smoother trajectory
+            return_full_result (bool): Whether to return the full result or not. If False, will randomly sample a single
+                (successful) trajectory to return
+            success_ratio (None or float): If set, specifies the fraction of successes necessary given self.batch_size.
+                If None, will automatically be the smallest ratio (1 / self.batch_size), i.e: any nonzero number of
+                successes
+            attached_obj (None or BaseObject): If specified, the object to attach to the robot end-effector when
+                solving for this trajectory
+
+        Returns:
+            None or th.tensor or MotionGenResult: If @return_all is set, will return the raw MotionGenResult object
+                computed from the batch trajectory computation. If not set, and there is at least a single successful
+                trajectory, will return a randomly sampled (N, D) tensor representing the interpolated joint trajectory
+                to reach the desired @target_pos, @target_quat configuration, where N is the number of interpolated
+                steps and D is the number of robot joints. If there is no successful trajectory, will return None.
```

Comment:
another reason to differentiate B, N, D, etc :D

Reply:
done